lib/internal/process/per_thread.js

```diff
@@ -181,8 +181,9 @@ function wrapProcessMethods(binding) {
   function exit(code) {
     process.off('exit', handleProcessExit);
 
-    if (code || code === 0)
+    if (code !== null && code !== undefined) {
```

Comment:
```suggestion
    if (code != null) {
```

Reply:
Fixed with `if (code !== undefined) {`. It doesn't seem to work in the following case. The exit status should be 1.

```console
$ ./node -e "process.exitCode=1; process.exit()"; echo $?                          
0
```